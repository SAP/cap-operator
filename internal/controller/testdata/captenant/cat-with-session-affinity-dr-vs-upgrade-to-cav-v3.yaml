apiVersion: sme.sap.com/v1alpha1
kind: CAPTenant
metadata:
  finalizers:
    - sme.sap.com/captenant
  generation: 2
  annotations:
    sme.sap.com/btp-app-identifier: btp-glo-acc-id.test-cap-01
    sme.sap.com/owner-identifier: default.test-cap-01
  labels:
    sme.sap.com/btp-app-identifier-hash: f20cc8aeb2003b3abc33f749a16bd53544b6bab2
    sme.sap.com/btp-tenant-id: tenant-id-for-provider
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: 1f74ae2fbff71a708786a4df4bb2ca87ec603581
    sme.sap.com/tenant-type: provider
  name: test-cap-01-provider
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPApplication
      name: test-cap-01
      uid: 3c7ba7cb-dc04-4fd1-be86-3eb3a5c64a98
spec:
  capApplicationInstance: test-cap-01
  subDomain: my-provider
  tenantId: tenant-id-for-provider
  version: 11.12.13
  versionUpgradeStrategy: always
status:
  observedGeneration: 2
  conditions:
    - message: "VirtualService (and DestinationRule) default.test-cap-01-provider was reconciled"
      observedGeneration: 2
      reason: TenantNetworkingModified
      status: "True"
      type: Ready
  state: Ready
  currentCAPApplicationVersionInstance: test-cap-01-cav-v3
  previousCAPApplicationVersions:
    - test-cap-01-cav-v1
    - test-cap-01-cav-v2
---
apiVersion: sme.sap.com/v1alpha1
kind: CAPTenantOperation
metadata:
  generateName: test-cap-01-provider-
  annotations:
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/cav-version: "11.12.13"
    sme.sap.com/tenant-operation-type: upgrade
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
  name: test-cap-01-provider-upg
  namespace: default
  finalizers:
    - sme.sap.com/captenantoperation
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  capApplicationVersionInstance: test-cap-01-cav-v3
  subDomain: my-provider
  tenantId: tenant-id-for-provider
  operation: upgrade
  steps:
    - name: mtx
      type: TenantOperation
status:
  conditions:
    - message: job default.test-cap-01-provider-upg-abcd completed
      reason: StepCompleted
      status: "True"
      type: Ready
  state: Completed
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    sme.sap.com/resource-hash: 0ff80461a6fee2f65b5e7ccd5243bec76c8db2b2bf330524fe7ab0911a7339ce
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
  name: test-cap-01-provider
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  gateways:
    - test-cap-01-primary-gen
    - default/test-cap-01-secondary-gen
  hosts:
    - my-provider.app-domain.test.local
    - my-provider.foo.bar.local
  http:
  - headers:
      response:
        set:
          Set-Cookie: COP_CAV=test-cap-01-cav-v1;Path=/;HttpOnly;Secure;Max-Age=0
    match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v1($|; .*)
      uri:
        regex: ^|.*(logout|logoff).*
    route:
    - destination:
        host: test-cap-01-cav-v1-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - headers:
      response:
        set:
          Set-Cookie: COP_CAV=test-cap-01-cav-v2;Path=/;HttpOnly;Secure;Max-Age=0
    match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v2($|; .*)
      uri:
        regex: ^|.*(logout|logoff).*
    route:
    - destination:
        host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v1($|; .*)
    route:
    - destination:
        host: test-cap-01-cav-v1-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v2($|; .*)
    route:
    - destination:
        host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - headers:
      response:
        add:
          Set-Cookie: COP_CAV=test-cap-01-cav-v2;Path=/;HttpOnly;Secure
    route:
    - destination:
        host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
---
apiVersion: networking.istio.io/v1
kind: DestinationRule
metadata:
  annotations:
    sme.sap.com/resource-hash: 97ecccd1443b2219a85159a5fdb2f8444543746a5ae890114462e6666e556696
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
  generation: 1
  name: test-cap-01-provider
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      consistentHash:
        httpCookie:
          name: CAPOP_ROUTER_STICKY
          path: /
          ttl: 0s
---
apiVersion: networking.istio.io/v1
kind: DestinationRule
metadata:
  annotations:
    sme.sap.com/resource-hash: 76ac6b80ce55711ae052011d8d29727030c897d4869ba6c403ac6842f08b93d6
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
  name: test-cap-01-provider-test-cap-01-cav-v1
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  host: test-cap-01-cav-v1-app-router-svc.default.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      consistentHash:
        httpCookie:
          name: CAPOP_ROUTER_STICKY
          path: /
          ttl: 0s
