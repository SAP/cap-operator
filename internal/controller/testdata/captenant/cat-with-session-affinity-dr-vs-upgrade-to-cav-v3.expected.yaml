apiVersion: sme.sap.com/v1alpha1
kind: CAPTenant
metadata:
  finalizers:
    - sme.sap.com/captenant
  generation: 2
  annotations:
    sme.sap.com/btp-app-identifier: btp-glo-acc-id.test-cap-01
    sme.sap.com/owner-identifier: default.test-cap-01
  labels:
    sme.sap.com/btp-app-identifier-hash: f20cc8aeb2003b3abc33f749a16bd53544b6bab2
    sme.sap.com/btp-tenant-id: tenant-id-for-provider
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: 1f74ae2fbff71a708786a4df4bb2ca87ec603581
    sme.sap.com/tenant-type: provider
  name: test-cap-01-provider
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPApplication
      name: test-cap-01
      uid: 3c7ba7cb-dc04-4fd1-be86-3eb3a5c64a98
spec:
  capApplicationInstance: test-cap-01
  subDomain: my-provider
  tenantId: tenant-id-for-provider
  version: 11.12.13
  versionUpgradeStrategy: always
status:
  observedGeneration: 2
  conditions:
    - message: "VirtualService (and DestinationRule) default.test-cap-01-provider was reconciled"
      observedGeneration: 2
      reason: TenantNetworkingModified
      status: "True"
      type: Ready
  state: Ready
  currentCAPApplicationVersionInstance: test-cap-01-cav-v3
  previousCAPApplicationVersions:
    - test-cap-01-cav-v1
    - test-cap-01-cav-v2
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    sme.sap.com/resource-hash: cea263f1fc91c820d166e30cab1dcf242966a7919090a3eb1a9907c72d596998
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
    sme.sap.com/tenant-type: provider
  name: test-cap-01-provider
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  gateways:
    - test-cap-01-primary-gen
    - default/test-cap-01-secondary-gen
  hosts:
    - my-provider.app-domain.test.local
    - my-provider.foo.bar.local
  http:
  - headers:
      response:
        set:
          Set-Cookie: COP_CAV=test-cap-01-cav-v2;Path=/;HttpOnly;Secure;Max-Age=0
    match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v2($|; .*)
      uri:
        regex: ^|.*(logout|logoff).*
    route:
    - destination:
        host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - headers:
      response:
        set:
          Set-Cookie: COP_CAV=test-cap-01-cav-v3;Path=/;HttpOnly;Secure;Max-Age=0
    match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v3($|; .*)
      uri:
        regex: ^|.*(logout|logoff).*
    route:
    - destination:
        host: test-cap-01-cav-v3-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v2($|; .*)
    route:
    - destination:
        host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - match:
    - headers:
        Cookie:
          regex: (^|.*; )COP_CAV=test-cap-01-cav-v3($|; .*)
    route:
    - destination:
        host: test-cap-01-cav-v3-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
  - headers:
      response:
        add:
          Set-Cookie: COP_CAV=test-cap-01-cav-v3;Path=/;HttpOnly;Secure
    route:
    - destination:
        host: test-cap-01-cav-v3-app-router-svc.default.svc.cluster.local
        port:
          number: 5000
      weight: 100
---
apiVersion: networking.istio.io/v1
kind: DestinationRule
metadata:
  annotations:
    sme.sap.com/resource-hash: 921a053e719ece9bc922497f98452ffd3ad4d057039fd831e3e46da92c4d2df4
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
    sme.sap.com/tenant-type: provider
  generation: 1
  name: test-cap-01-provider
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  host: test-cap-01-cav-v3-app-router-svc.default.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      consistentHash:
        httpCookie:
          name: CAPOP_ROUTER_STICKY
          path: /
          ttl: 0s
---
apiVersion: networking.istio.io/v1
kind: DestinationRule
metadata:
  annotations:
    sme.sap.com/resource-hash: 97ecccd1443b2219a85159a5fdb2f8444543746a5ae890114462e6666e556696
    sme.sap.com/owner-identifier: default.test-cap-01-provider
  labels:
    sme.sap.com/owner-generation: "2"
    sme.sap.com/owner-identifier-hash: db1f1fd7eaeb0e6407c741b7e4b2540044bcc4ec
    sme.sap.com/tenant-type: provider
  name: test-cap-01-provider-test-cap-01-cav-v2
  namespace: default
  ownerReferences:
    - apiVersion: sme.sap.com/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: CAPTenant
      name: test-cap-01-provider
spec:
  host: test-cap-01-cav-v2-app-router-svc.default.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      consistentHash:
        httpCookie:
          name: CAPOP_ROUTER_STICKY
          path: /
          ttl: 0s
